### PR TITLE
Implements Sticky and Progress flashes

### PR DESCRIPTION
**Pull request by discorick**
This should give a lot more flexibility in the app over how we handle notifications with the flash component. Gifs below.

### CODE REVIEW

File: app/assets/stylesheets/components/_flash-message.scss [hunk 1]
```diff
@@ -8,12 +8,26 @@
...
     margin-bottom: 4px;
     box-shadow: -1px 2px 2px rgba(0,0,0,0.5);
   }
+
+  .info {
+    background: $hb-purple-dark;
+  }
+
+  .warning {
+    background: #f9646e;
```
Comment (dahlbyk) [inline app/assets/stylesheets/components/_flash-message.scss]: Variable?

File: ember-app/app/components/flash/hb-message.js [hunk 2]
```diff
@@ -2,6 +2,14 @@ import Ember from 'ember';
 
 var  HbFlashComponent = Ember.Component.extend({
   classNames: ['message'],
+  classNameBindings: ['flash.id'],
+  progress: Ember.computed.alias('flash.progress.status'),
+
+  click: function(){
+    if(this.get('progress'){return;})
```
Comment (dahlbyk) [inline ember-app/app/components/flash/hb-message.js]: Is closing paren here misplaced?

**Comment by dahlbyk** [comment]
I think we're good here. Anything else you want to test, or anything you specifically want me to try out locally?

**Comment by discorick** [comment]
Just want to do a little bit more testing, just to make sure nothing in the current production logic regressed.

**Comment by discorick** [comment]
Progress Indicator Flash:
untitled recording

Sticky Warning Flash:
untitled recording 1

**Comment by dahlbyk** [comment]
I am content. :shipit:?

**Comment by discorick** [comment]
With x for close:

image

**Comment by dahlbyk** [comment]
> With x for close:

Can the x be aligned with the first message row's text?
firefox_2016-04-14_00-23-21

**Comment by discorick** [comment]
When the message overflows to an additional row the x starts to look pretty misplaced when its inline.

**Comment by dahlbyk** [comment]
> Keeps the sticky toast x from overlapping with long messages

What does it look like now?

**Comment by discorick** [comment]
Can we consider this 'good enough' so I can rebase it into: #278 ?

**Comment by dahlbyk** [comment]
> Can we consider this 'good enough' so I can rebase it into: #278 ?

:+1:

**Pull request merged**